sdk/core/Azure.Core/src/Pipeline/RequestFailedDetailsParser.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Core
+{
+    /// <summary>
+    ///
+    /// </summary>
+    public abstract class RequestFailedDetailsParser
+    {
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="response"></param>
+        /// <returns></returns>
+        public abstract RequestFailedDetails GetRequestFailedDetails(Response response);
```

Comment:
There is no reason to implement this class if you don't need to override this method. Is there a use case you had in mind where we'd want to call the base implementation?

Reply:
I think it's worth considering @JoshLove-msft's suggestion.  What would it look like if instead of checking the parser for null, we just always called the parser, and if no one had subclassed it, we got the default implementation?  ResponseClassifier uses this design, so there's a precedent for it in the API.  Does it make things simpler, or more complex?

Reply:
Ideally, I agree with the idea that we should always call the parser rather than checking for null. My main concern with doing that is it relies on the `HttpPipelineTransportPolicy` which will almost always be there but is not guaranteed to be there. My thought was that we should be erring on the side of avoiding a NRE or failing to call even the default parser. 

But the main concern is just that making it virtual doesn't force sub-classes to provide an implementation at all, which is really the only reason one would sub-class the type.

Reply:
> My concern with doing that is it relies on the HttpPipelineTransportPolicy

Can you help me understand this depencency?

> But the main concern is just that making it virtual doesn't force sub-classes to provide an implementation at all, which is really the only reason one would sub-class the type.

Do we need to design for the case where someone wants to subclass a type with one virtual method and not override the virtual method?  I don't understand why someone would do that.

Reply:
discussed offline - some takeaways:
- There is no reason to implement a sub-class of this type unless one intends to do custom error parsing
- There are no current examples of custom `ClientDiagnostics.ExtractFailureContent` implementations that call the default implementation as part of their custom implementation
- Regardless of whether this is purely abstract or virtual, the client author code that is written is identical. 
- question for the architects: Is the existence of a pure abstract type for client authors to implement inconsistent with our current APIs? If so, how much should this inconsistency be weighed against the use case?